test/integration/segmentation_camera.cc

```diff
@@ -0,0 +1,262 @@
+                    unsigned int /*_channels*/,
+                    const std::string &/*_format*/)
+{
+  g_mutex.lock();
+  auto bufferSize = _width * _height * 3;
+
+  if (!g_buffer)
+    g_buffer = new uint8_t[bufferSize];
+
+  memcpy(g_buffer, _data, bufferSize);
+
+  g_counter++;
+  g_mutex.unlock();
+}
+
+/// \brief Build the scene with 3 boxes besides each other
+/// the 2 aside boxes have the same label & the middle is different
+void BuildScene(rendering::ScenePtr scene)
+{
+  math::Vector3d leftPosition(3, -1.5, 0);
+  math::Vector3d rightPosition(3, 1.5, 0);
+  math::Vector3d middlePosition(3, 0, 0);
+
+  rendering::VisualPtr root = scene->RootVisual();
+
+  // create box visual
+  rendering::VisualPtr box = scene->CreateVisual();
+  box->AddGeometry(scene->CreateBox());
+  box->SetOrigin(0.0, 0.0, 0.0);
+  box->SetLocalPosition(leftPosition);
+  box->SetLocalRotation(0, 0, 0);
+  box->SetUserData("label", 1);
+  root->AddChild(box);
+
+  // create box visual of same label
+  rendering::VisualPtr box1 = scene->CreateVisual();
+  box1->AddGeometry(scene->CreateBox());
+  box1->SetOrigin(0.0, 0.0, 0.0);
+  box1->SetLocalPosition(rightPosition);
+  box1->SetLocalRotation(0, 0, 0);
+  box1->SetUserData("label", 1);
+  root->AddChild(box1);
+
+  // create box visual of different label
+  ignition::rendering::VisualPtr box2 = scene->CreateVisual();
+  box2->AddGeometry(scene->CreateBox());
+  box2->SetOrigin(0.0, 0.0, 0.0);
+  box2->SetLocalPosition(middlePosition);
+  box2->SetLocalRotation(0, 0, 0);
+  box2->SetUserData("label", 2);
+  root->AddChild(box2);
+}
+
+//////////////////////////////////////////////////
+void SegmentationCameraTest::SegmentationCameraBoxes(
+  const std::string &_renderEngine)
+{
+  // Optix is not supported
+  if (_renderEngine.compare("optix") == 0)
+  {
+    igndbg << "Engine '" << _renderEngine
+              << "' doesn't support depth cameras" << std::endl;
+    return;
+  }
+
+  // Setup ign-rendering with an empty scene
+  auto *engine = ignition::rendering::engine(_renderEngine);
+  if (!engine)
+  {
+    igndbg << "Engine '" << _renderEngine
+              << "' is not supported" << std::endl;
+    return;
+  }
+
+  ignition::rendering::ScenePtr scene = engine->CreateScene("scene");
+  BuildScene(scene);
+
+  // Create Segmentation camera
+  auto camera = scene->CreateSegmentationCamera("SegmentationCamera");
+  ASSERT_NE(camera, nullptr);
+
+  camera->SetLocalPosition(0.0, 0.0, 0.0);
+  camera->SetLocalRotation(0.0, 0.0, 0.0);
+
+  int backgroundLabel = 23;
+  camera->SetBackgroundLabel(backgroundLabel);
+  EXPECT_EQ(camera->BackgroundLabel(), backgroundLabel);
+
+  camera->SetSegmentationType(SegmentationType::ST_SEMANTIC);
+
+  EXPECT_EQ(camera->Type(), SegmentationType::ST_SEMANTIC);
+
+  camera->EnableColoredMap(false);
+  EXPECT_FALSE(camera->IsColoredMap());
+
+  int width = 320;
+  int height = 240;
+  double aspectRatio = width/height;
```

Comment:
```suggestion
  double aspectRatio = static_cast<double>(width) / static_cast<double>(height);
```
In order to get proper division here, the ints need to be casted to doubles before dividing. Otherwise, integer division will occur, resulting in `1`, and then `1` will be converted to a double when assigned to `aspectRatio`. Here's an example of what I'm describing: https://godbolt.org/z/6dcMKqhrW